app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPromptViewModel.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.newMessage
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.infomaniak.mail.data.api.ApiRepository
+import com.infomaniak.mail.di.IoDispatcher
+import com.infomaniak.mail.utils.coroutineContext
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AiPromptViewModel @Inject constructor(
+    application: Application,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) : AndroidViewModel(application) {
+
+    private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
+
+    val aiProposition = MutableLiveData<String>()
+
+    fun generateAiProposition(prompt: String) = viewModelScope.launch(ioCoroutineContext) {
+        with(ApiRepository.generateAiProposition(prompt)) {
```

Comment:
I thought we stopped doing with() like this on ApiResponse, to see with @KevinBoulongne

Reply:
We are still doing it.